eng/common/tools.ps1

```diff
@@ -436,7 +442,7 @@ function MSBuild() {
   $cmdArgs = "$($buildTool.Command) /m /nologo /clp:Summary /v:$verbosity /nr:$nodeReuse"
 
   if ($warnAsError) { 
-    $cmdArgs += " /p:TreatWarningsAsErrors=true" 
+    $cmdArgs += " /warnaserror /p:TreatWarningsAsErrors=true" 
```

Comment:
@sharwell as I believe we removed this for the formatting analyzer work.

Reply:
We can't be modifying tools.ps1 in Roslyn. If we need to disable warnings as errors then we need to set `$warnAsError = $false` in `build.ps1`

Reply:
Forgot this was a protected directory. Actually I think this is fine. We don't need to differentiate between warning and error anymore. The packages around code style are completely disabled in the specific runs where we care about this.

Reply:
@tmat Build in master is failing until this change is reverted.